url294e6c45f3_Ecdosis_NMergeNew_tgz-pJ8-TestJ8/src/MvdTool.java

```diff
@@ -0,0 +1,975 @@
+						+"\"my new MVD\" -e UTF-8");
+					break;
+				case DESCRIPTION:
+					out.println( "nmerge -c description -m work.mvd\n"
+						+"nmerge -c description -m work.mvd -d \"my "
+						+"new mvd description\"");
+					break;
+				case TREE:
+					out.println( "nmerge -c tree -m work.mvd -d existing.mvd" );
+					break;
+				case EXPORT:
+					out.println( "nmerge -c export -m work.mvd "
+						+"-x work.xml");
+					break;
+				case FIND:
+					out.println( "nmerge -c find -m work.mvd -f \"bananas are nice\"");
+					break;
+				case IMPORT:
+					out.println( "nmerge -c import -m work.mvd -x work.xml");
+					break;
+				case LIST:
+					out.println( "nmerge -c list -m work.mvd");
+					break;
+				case READ:
+					out.println( "nmerge -c read -m work.mvd -v 4");
+					break;
+				case UNARCHIVE:
+					out.println( "nmerge -c unarchive -m work.mvd "+
+						"-a myArchiveFolder");
+					break;
+				case UPDATE:
+					out.println( "nmerge -c update -m work.mvd -v 3 -t 5.txt");
+					break;
+				case VARIANTS:
+					out.println( "nmerge -c variants -m work.mvd "
+						+"-v 3 -o 1124 -k 23");					
+					break;
+                case WORDS:
+                    out.println( "nmerge -c words -m work.mvd" );
+                    break;
+				case HELP: case USAGE:
+					usage();
+					break;
+				default:
+					throw new Exception( "Command "+hComm+" not catered for " );
+			}
+		}
+		catch ( Exception e )
+		{
+			throw new MVDToolException( e );
+		}
+	}
+	/**
+	 * Compare two versions
+	 * @throws MVDToolException
+	 */
+	private static void doCompare() throws MVDToolException
+	{
+		try
+		{
+			MVD mvd = loadMVD();
+			if ( version == 0 || with == 0 )
+				throw new MVDToolException( 
+					"can't compare version="+version+" with version "+with );
+			Chunk[] chunks = mvd.compare( version, with, uniqueState );
+			if ( mvd.getEncoding().toUpperCase().equals("UTF-8") )
+                out.write( UTF8_BOM, 0, UTF8_BOM.length );
+            for ( int i=0;i<chunks.length;i++ )
+			{
+				char[] chars = chunks[i].getChars();
+				out.print( chars );
+			}
+		}
+		catch ( Exception e )
+		{
+			throw new MVDToolException( e );
+		}
+	}
+	/**
+	 * Load an MVD file from the specified path
+	 * @return the MVD, loaded
+	 */
+	static MVD loadMVD() throws Exception
+	{
+		MVD mvd;
+		File m = new File( mvdFile );
+		if ( m.exists() || (dbConn != null && dbConn.length()>0) )
+			mvd = MVDFile.internalise( m, Utilities.loadDBProperties(dbConn) );
+		else
+			throw new FileNotFoundException( "Couldn't find "+mvdFile );
+        mvd.setDirectAlign( directAlignOnly );
+		return mvd;
+	}
+	/**
+	 * Update an MVD by replacing the specified version with the given 
+	 * textfile
+	 */
+	private static void doUpdateMVD() throws MVDToolException
+	{
+		try
+		{
+			File t = new File( textFile );
+			if ( t.exists() )
+			{
+				MVD mvd = loadMVD();
+				mvd.removeVersion( version );
+				FileInputStream fis = new FileInputStream( t );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *fis*. Execution paths that do not contain closure statements were detected. Ensure *fis* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).